### DISCUSSION TITLE
Supporting custom lint rules

### DISCUSSION THREAD

**Discussion by nfichter** [biomejs/biome#231] [Ideas]
A feature of ESLint my organization used before migrating to Biome was writing our own custom ESLint rules (e.g. automatically replacing imports from a certain package with our own custom wrappers around those functions). Is there a philosophical reason against Biome supporting custom lint rules or is it just pending someone implementing it? I'm happy to work on it, just wanted to check and make sure before I get started.

**Comment by Conaclos** [comment]
There is no particular philosophical reason that I know of. We haven't taken the time to investigate how to support them efficiently. I think this requires some investigation before we move towards implementation. We are open to suggestions and time investment from new contributors :)

**Reply by nfichter** [reply]
Makes sense. Are there any specific efficiency concerns off the top of your head? I'm still familiarizing myself with the codebase, so I'm not sure about any hypothetical implementation details yet, but happy to spend some time investigating + contributing!

**Reply by leonard-henriquez** [reply]
We also have custom rules in my company. That's what prevents us from moving to Biome but we definitely would if we had a way to create our custom rules like in ESLint !
We are very interested if you plan to work on it @nfichter or @Conaclos

**Reply by ematipico** [reply]
> We also have custom rules in my company. That's what prevents us from moving to Biome but we definitely would if we had a way to create our custom rules like in ESLint !
> We are very interested if you plan to work on it @nfichter or @Conaclos 

Just a suggestion: you can use Biome, and eslint for your custom rules. 

Sentry has found some interesting numbers: https://github.com/getsentry/sentry/pull/61397

If CI time is not a concern in your company, you can disregard my suggestion.

**Reply by leonard-henriquez** [reply]
Performance is definitely not great with eslint (our eslint takes more than 5 minutes to run on big instances), but since it runs in parallel with our CI it's not really painful.
I was appealed by Biome by the "One toolchain for your web project" tagline

**Reply by IbrahimShamma99** [reply]
> Just a suggestion: you can use Biome, and eslint for your custom rules.

We will be parsing the same file twice, not optimal for most projects I think

**Comment by cristianpoleyJS** [comment]
I jump here because I am also interested in using Biome as a tool for everything (linter and formatter) but in my organization we also have custom eslint rules. Any news regarding this? thanks!

**Reply by ematipico** [reply]
https://github.com/biomejs/biome/discussions/1649

**Comment by arendjr** [comment]
I think it’s best to close this discussion and direct interested users to our upcoming plugin efforts instead: https://github.com/biomejs/biome/discussions/1762
